.circleci/config.yml

```diff
@@ -1,31 +1,116 @@
+  node_image: &node_image
+    image: circleci/node:8
+  ipfs_image: &ipfs_image
+    image: ipfs/go-ipfs
+  ganache_image: &ganache_image
+    image: trufflesuite/ganache-cli
+    command:
+      - "-l"
+      - "90000000"
+      - "-m"
+      - "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat"
+
 jobs:
   build:
     docker:
-      - image: circleci/node:8.11.3-stretch
-      - image: ipfs/go-ipfs:v0.4.16
-      - image: trufflesuite/ganache-cli:v6.1.6
-        command: [
-          "-l", "90000000",
-          "-m", "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat"]
-    environment:
-      NODE_ENV: test
+      - *node_image
+    working_directory: *working_directory
     steps:
       - checkout
-      - restore_cache:
-          key: v1-node-modules-cache-{{ checksum "package.json" }}-{{ checksum "packages/requestNetwork.js/package.json" }}
-      - run:
-          name: Install depedencies
-          command: |
-            npm install .
-            cd packages/requestNetwork.js/ && npm install . && cd -
-      - save_cache:
-          key: v1-node-modules-cache-{{ checksum "package.json" }}-{{ checksum "packages/requestNetwork.js/package.json" }}
-          paths:
-            - "node_modules"
-            - "packages/requestNetwork.js/node_modules"
-      - run:
-          name: Deploy test contracts and run Request unit tests
-          command: |
-            ./node_modules/lerna/bin/lerna.js run --scope @requestnetwork/request-network.js testdeploy
-            ./node_modules/lerna/bin/lerna.js run --scope @requestnetwork/request-network.js test --stream
+      - run:
+          name: 'Install lerna'
+          command: 'sudo npm install -g lerna'
+      - run:
+          name: 'Bootstrap'
+          command: 'lerna bootstrap --concurrency=1'
+      - persist_to_workspace:
+          root: *working_directory
+          paths: .
+  test-library:
+    docker:
+      - *node_image
+      - *ipfs_image
+      - *ganache_image
+    working_directory: *working_directory
+    steps:
+      - attach_workspace:
+          at: *working_directory
+      - run:
+          name: 'Install lerna'
+          command: 'sudo npm install -g lerna'
+      - run:
+          name: 'Deploy test contract'
+          command: 'lerna run --scope @requestnetwork/request-network.js testdeploy'
+      - run:
+          name: 'Test requestNetwork.js library'
+          command: 'lerna run --scope @requestnetwork/request-network.js test --stream'
+  test-contract:
+    docker:
+      - *node_image
+      - *ipfs_image
+      - *ganache_image
+    working_directory: *working_directory
+    steps:
+      - attach_workspace:
+          at: *working_directory
+      - run:
+          name: 'Install lerna'
+          command: 'sudo npm install -g lerna'
+      - run:
+          name: 'Deploy test contract'
+          command: 'lerna run --scope @requestnetwork/request-network.js testdeploy'
```

Comment:
Tests for request-network-smart-contracts don't require to deploy the contracts with `testdeploy`